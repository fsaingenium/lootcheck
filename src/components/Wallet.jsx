import React, { Component } from "react";
import { connect } from "react-redux";
import { deposit, withdraw } from "../actions/balance";

export class Wallet extends Component {
  constructor(props) {
    super(props);

    this.state = {
      balance: undefined
    };
  }

  updateBalance = event => {
    this.setState({ balance: parseInt(event.target.value, 10) });
  };

  deposit = () => this.props.deposit(this.state.balance);

  withdraw = () => this.props.withdraw(this.state.balance);

  render() {
    const { balance } = this.props;
    return (
      <div>
        <h3 className="balance">Wallet balance: {balance}</h3>
        <br />
        <input
          type="text"
          className="input-wallet"
          onChange={this.updateBalance}
        />
        <button className="btn-deposit" onClick={this.deposit}>
          Deposit
        </button>
        <button className="btn-withdraw" onClick={this.withdraw}>
          Withdraw
        </button>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  balance: state
});

const mapDispatchToProps = dispatch => ({
  deposit: value => dispatch(deposit(value)),
  withdraw: value => dispatch(withdraw(value))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Wallet);
