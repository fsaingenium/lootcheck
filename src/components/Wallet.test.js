import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Wallet } from "./Wallet";

Enzyme.configure({ adapter: new Adapter() });

describe("Wallter", () => {
  const mockDeposit = jest.fn();
  const mockWithdraw = jest.fn();
  const props = { balance: 20, deposit: mockDeposit, withdraw: mockWithdraw };
  const wallet = shallow(<Wallet {...props} />);

  it("renders correctly", () => {
    expect(wallet).toMatchSnapshot(wallet);
  });

  it("display the balance from props", () => {
    expect(wallet.find(".balance").text()).toEqual("Wallet balance: 20");
  });

  it("creates an input to withdraw or deposit from the balance", () => {
    expect(wallet.find(".input-wallet").exists()).toBe(true);
  });

  describe("when the user types into the wallet input", () => {
    const userBalance = "25";

    beforeEach(() => {
      wallet
        .find(".input-wallet")
        .simulate("change", { target: { value: userBalance } });
    });

    it("update the value in the `state` end converts it to a number", () => {
      expect(wallet.state().balance).toEqual(parseInt(userBalance, 10));
    });

    describe("and the user wants to make a deposit", () => {
      beforeEach(() => {
        wallet.find(".btn-deposit").simulate("click");
      });

      it("dispatches the `deposit()` it recieves from props with local balance", () => {
        expect(mockDeposit).toHaveBeenCalledWith(parseInt(userBalance, 10));
      });
    });

    describe("and the user wants to make a withdraw", () => {
      beforeEach(() => {
        wallet.find(".btn-withdraw").simulate("click");
      });

      it("dispatches the `withdraw()`...", () => {
        expect(mockWithdraw).toHaveBeenCalledWith(parseInt(userBalance, 10));
      });
    });
  });
});
