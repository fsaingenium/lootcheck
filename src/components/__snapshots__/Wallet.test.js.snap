// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallter renders correctly 1`] = `Object {}`;
